smart-contracts/contracts/mixins/MixinKeys.sol

```diff
@@ -148,7 +148,16 @@ contract MixinKeys is
     public
     view
     returns (bool)
-  {
+  { 
+    // use hook if it exists
+    if(address(onBalanceOfHook) != address(0)) {
```

Comment:
So I think a more powerful approach would be to still have 
`balanceOf = keyByOwner[_keyOwner].expirationTimestamp > block.timestamp;` and then, if the hook is defined, call it with all the params you currently have AND `balanceOf`, so that the 3rd party hook can decide on its logic completely.